src/wireviz/DataClasses.py

```diff
@@ -55,6 +59,18 @@ def __post_init__(self):
         if len(self.pinnumbers) != len(set(self.pinnumbers)):
             raise Exception('Pin numbers are not unique')
 
+        if self.show_name is None:
+            if self.autogenerate:
+                self.show_name = False  # hide auto-generated designators by default
+            else:
+                self.show_name = True
```

Comment:
The if-else spanning lines 63-66 might be replaced by:
`self.show_name = not self.autogenerate  # hide auto-generated designators by default`

Reply:
Done.